workflow-runtime/src/commonMain/kotlin/com/squareup/workflow1/internal/WorkflowNode.kt

```diff
@@ -205,19 +276,34 @@ internal class WorkflowNode<PropsT, StateT, OutputT, RenderingT>(
   ) {
     if (newProps != lastProps) {
       val newState = interceptor.intercept(workflow, this)
-        .onPropsChanged(lastProps, newProps, state)
-      state = newState
+        .onPropsChanged(lastProps, newProps, state.value)
+      state.value = newState
     }
     lastProps = newProps
   }
 
+  @Composable
+  private fun UpdatePropsAndState(
+    workflow: StatefulWorkflow<PropsT, StateT, OutputT, RenderingT>,
+    newProps: PropsT
+  ) {
+    if (newProps != lastProps) {
+      val newState = interceptor.intercept(workflow, this)
+        .onPropsChanged(lastProps, newProps, state.value)
```

Comment:
I hope everyone heeded our advice to not perform side effects in this function 😅

Reply:
I think we can protect against that anyway like this:

```
    val newState = produceState(initialValue = state.value, newProps) {
      if (newProps != lastProps) {
        interceptor.intercept(workflow, this@WorkflowNode)
          .onPropsChanged(lastProps, newProps, state.value)
      }
    }
    state.value = newState.value
```